spec/classes/trusted_facts_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe 'trusted_facts', :if => Puppet.version.to_f >= 4.0 do
```

Comment:
Given the test results, and the fact that 4.0, and 4.1 are not supported anymore, you can push this check up to `>= 4.3` and mention this in the readme.